### ISSUE TITLE
Peer Review - Feedback

### ISSUE THREAD

**Issue by shlrley** [UBC-MDS/maternal_health_risk_predictor#40]
**Reviewer 1:** 
- [ ] The decision to go with DecisionTreeClassifier is not comprehensive. Perhaps the authors can first discuss what each model does in making predictions and then conclude why DecisionTreeClassifier is an appropriate estimator compared to SVM, for example.
- [ ] Analysis of the results is somewhat incomplete. The authors mentioned in Future Directions that they can set the classes as binary by merging high-medium or medium-low risk groups. Is it possible to do this before the final report and thereby improve the model by setting precision, recall, or f1-score as the scoring metric?
- [ ] If choosing another scoring metric, which score should the model maximize, and the reason behind it?
- [ ] The background on the risk levels is lacking. I understand that this might be out of the expertise of the authors, but please indicate any resource to help readers understand what high, medium, and low risk indicate. For example, if a woman has a medium risk of maternal mortality, what does it mean?
- [ ] After the EDA section, can the authors explain any decision made to transform the data? (Standardizing, imputing, or any unit conversions)?
- [ ] For Figure 3 shows the correlation, it's very difficult to tell which groups have a correlation or not. Perhaps the authors can include a table showing the correlation coefficients to help in interpretation.

**Comment by shlrley** [comment]
**Reviewer 2:**
- [ ] The authors can consider making all their scripts better documented (for example, in download_data.py, add the docstring to the main function you defined; in pre_processing.py, add comments on your code for each part, like # test # export data...).
- [ ] For the Contribute guidelines, the authors can consider adding methods for others to report the software issues or other problems.
- [ ] For the EDA Figure 3, the authors can only show the plots for the features SystolicBP and DiastolicBP with high correlation compared to other pairs of predictors and remove the insignificant correlation plots to make it easier for viewers to read.
- [ ] For each script, the authors can consider removing unused imported modules within that script, like ConfusionMatrixDisplay in fit_maternal_risk_predict_model.py ,and make sure all modules do what they should do. (not sure why the authors comment out the code for create_confusionmatrix in fit_maternal_risk_predict_model.py)
- [ ] Given the data size is relatively small (<1000 observations for train_df), the authors might consider addressing this in the limitation of the analysis part.
- [ ] The overall results seem unfinished since the authors comment out the code for create_confusionmatrix and scores on X_test in fit_maternal_risk_predict_model.py. The authors show the mean CV validation test score 0.823 on the best optimized decisiontree model, but how about the overall score on the TEST data? Without knowing these, we might have a big overfitting problem and the optimized decisiontree model may not generalize well in practice.
- [ ] As the authors mention in their future direction to use other score metrics, here I would suggest maybe first before in milestone 4, at least compare one other score metrics (f1-score is good) with the accuracy metrics you currently use.
- [ ] The authors can further consider justifying the reasons why using the selected features and the decisiontree model.

**Comment by shlrley** [comment]
**Reviewer 3:**

- [ ] This project is well put together and it is obvious that the group understands the typical data science workflows. It's worth noting that this group only has 3 members instead of 4, but this has not diminished the quality of the submission. I was specifically impressed by the quality of the code across the board. Clear, modular, and easy to follow, plus error handling and function testing! Excellent work! I also appreciate that the group chose a dataset with the potential to bring good to the world - I think it's important to make time to explore data like this when possible.
- [ ] I struggled to get the automated scripts to run. Unfortunately, download_data.py, eda_script.py, and rendering the final report all failed to run successfully on my Linux Ubuntu machine, even after installing all the listed packages in a new conda environment. I copy pasted directly from the usage section, but did not troubleshoot further if they did not work. I've included the command I ran and any console replies at the end of this comment if you'd like to investigate further.
- [ ] Regarding your conclusions, I would challenge your assertion that the Decision Tree Classifier is the best model to use here, despite it getting the best cross-validation scores. Taking 6 features to a depth of 29 means we're revisiting the same feature a lot and I suspect the model might suffer from overfitting as a result. There was a large difference between training and validation scores on your decision tree classifier. I was more inclined to choose the SVM model which did not suffer from the same overfitting, even though it cost you some accuracy. Also, as you discussed, we'd probably want to choose a different scoring function anyways which might change the equation again.

_Other thoughts I had while reviewing - feel free to take 'em or leave 'em :)_

- [ ] I like the density distribution plots in your EDA a lot. They clearly show that the distribution differs between the classes for almost every feature; you should be able to train a strong classifier as a result.
- [ ] You should be careful about making claims about the application of your model and findings to women's health in general because the sample is limited to a few years in a specific geographic region (rural Bangladesh).
- [ ] In the report, you mention "The R programming language was used to perform the analysis", but all of your analysis is done in python as far as I can see.
- [ ] Somewhere in your report I would have liked to see some indication on the severity of each risk level. How bad is "high", for example. How much worse is high than "mid".
- [ ] Because these are health related outcomes, you might want to choose a scoring metric that penalizes missing high risk patients more; better to end up with a "overly pessimistic" model than an "optimistic" one.
- [ ] You have the right idea in the "Future Directions" section, but keep in mind that we can maximize the "recall" of high risk patients by training a dummy model that always predicts "high risk". This is why f1 score is usually the alternative if we're worried about class imbalance and the type 2 errors for a particular class.
- [ ] Your "standard" classification model doesn't take into account the ordinality of the classes. It is more of a mistake to misclassify a High Risk patient as Low Risk than Medium Risk, and if this happens often (it doesn't on the test set - those are some big red flags for the model). This application might call for a custom scoring function.
- [ ] Alternatively, you might even consider encoding say 'Low Risk' to 0, 'Med Risk' to 0.5, and 'High Risk' to 1, then training a regression model instead. Obviously since the data isn't set up this way tread with caution, but I like the idea of a continuous prediction scale instead of a discrete one.
